compiler/passes/parallel.cpp

```diff
@@ -322,9 +322,93 @@ bundleArgs(CallExpr* fcall, BundleArgsFnData &baData) {
     i++;
   }
 
+
+
+  // Put the bundle into a void* argument
+  VarSymbol *allocated_args = newTemp(astr("_args_vfor", fn->name), dtCVoidPtr);
+  fcall->insertBefore(new DefExpr(allocated_args));
+  fcall->insertBefore(new CallExpr(PRIM_MOVE, allocated_args, tempc));
+
+  // Put the size of the bundle into the next argument
+  VarSymbol *tmpsz = newTemp(astr("_args_size", fn->name),
+                             dtInt[INT_SIZE_DEFAULT]);
+  fcall->insertBefore(new DefExpr(tmpsz));
+  fcall->insertBefore(new CallExpr(PRIM_MOVE, tmpsz,
+                                   new CallExpr(PRIM_SIZEOF, ctype->symbol)));
+
+  // Find the _EndCount argument so we can pass that explicitly as the
+  // first argument to a task launch function.
+  Symbol* endCount = NULL;
+  VarSymbol *taskList = NULL;
+  VarSymbol *taskListNode = NULL;
+
+  if (!fn->hasFlag(FLAG_ON)) {
+    for_actuals(arg, fcall) {
+      bool strcmp_found = false;
+      bool type_found = false;
+
+      Type* baseType = arg->getValType();
+      if (baseType->symbol->hasFlag(FLAG_END_COUNT)) {
```

Comment:
It doesn't look like FLAG_END_COUNT is ever added to anything. Is it worthwhile to include this check in this PR?